File: src/main/java/net/zenxarch/bot/command/PortalCommand.java
package net.zenxarch.bot.command;

import com.mojang.brigadier.CommandDispatcher;
import com.mojang.brigadier.context.CommandContext;
import net.fabricmc.fabric.api.client.command.v1.ClientCommandManager;
import net.fabricmc.fabric.api.client.command.v1.FabricClientCommandSource;
import net.minecraft.command.argument.BlockPosArgumentType;

public class PortalCommand {

  public static void register(
      CommandDispatcher<FabricClientCommandSource> dispatcher) {
    dispatcher.register(ClientCommandManager.literal("zportal").then(
        ClientCommandManager.argument("pos", BlockPosArgumentType.blockPos())
            .executes(ctx -> execute(ctx))));
  }

  private static int execute(CommandContext<FabricClientCommandSource> ctx) {
    // BlockPos pos = ctx.getArgument("pos", PosArgument.class);
    // mod.portalProcess.activate(pos);
    return 0;
  }
}

File: src/main/scala/net/zenxarch/bot/command/PortalCommand.scala
package net.zenxarch.bot.command

import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.context.CommandContext
import net.fabricmc.fabric.api.client.command.v1.ClientCommandManager._
import net.fabricmc.fabric.api.client.command.v1.FabricClientCommandSource as fs
import net.minecraft.command.argument.BlockPosArgumentType

class PortalCommand extends ZenCommand:
  def register() = literal("zportal")
    .`then`(
      argument("pos", BlockPosArgumentType.blockPos())
        .executes(this.execute)
    )

  private def execute(ctx: CommandContext[fs]): Int =
    // BlockPos pos = ctx.getArgument("pos", PosArgument.class)
    // mod.portalProcess.activate(pos)
    return 0
